### PR TITLE
#184: Sbt cross-build is broken

### LINKED ISSUE #184: Sbt cross-build is broken
## Describe the bug
The project uses sbt-matrix plugin for cross build. But git actions use ++ notations.
Also some combinations seem to fail to build.

Depends on #193

### PR DESCRIPTION
**Pull request by benedeki**
* build workflow script reworked to build, execute unit tests, execute integration tests and create code documentation
* JaCoCo report workflow script put into one file and fixed to work
* several other workflow scripts adjusted for easier reading by adding or adjusting job or file names

Closes #184

### CODE REVIEW

File: .github/workflows/jacoco_report.yml [hunk 1]
```diff
@@ -78,6 +101,8 @@ jobs:
           echo "Changed Files coverage ${{ steps.jacoco-agent.outputs.coverage-changed-files }}"
           echo "Total model module coverage ${{ steps.jacoco-model.outputs.coverage-overall }}"
           echo "Changed Files coverage ${{ steps.jacoco-model.outputs.coverage-changed-files }}"
+          echo "Total sever module coverage ${{ steps.jacoco-server.outputs.coverage-overall }}"
```
Comment (lsulak) [inline .github/workflows/jacoco_report.yml:104]:
```suggestion
          echo "Total server module coverage ${{ steps.jacoco-server.outputs.coverage-overall }}"
```

**Review by lsulak — COMMENTED** [review]
LGTM so far!  We'll need to adjust the required checks on the Repository Settings level as well, because these are not dynamically discovered as the code changes. I can help if you want.

**Comment by miroslavpojer** [comment]
See 6 expected checks which are visible. It should be removed.

**Comment by miroslavpojer** [comment]
In README.md - chapter How To Run Tests, there are misleading examples with the latest changes.
There is needed to add information on what Java version is needed where + adding "project XYZ" to the command examples.

**Review by miroslavpojer — COMMENTED** [review]
- [x] pulled
- [x] code review
- [x] local machine test

During review I have created 2 post not related to changed files.

File: .github/workflows/jacoco_report.yml [hunk 2]
```diff
@@ -28,48 +28,71 @@ env:
...
+      - name: Add coverage to PR (model)
+        if: steps.jacocorun.outcome == 'success'
+        id: jacoco-model
+        uses: madrapps/jacoco-report@v1.6.1
+        with:
+          name: model-jacoco-report
+          paths: ${{ github.workspace }}/model/target/jvm-${{ env.scalaShort }}/jacoco/report/jacoco.xml
+          token: ${{ secrets.GITHUB_TOKEN }}
+          min-coverage-overall: ${{ env.overall }}
+          min-coverage-changed-file: ${{ env.changed }}
```
Comment (TebaleloS) [inline .github/workflows/jacoco_report.yml:71]: Just like Ladislav said, I think this `min-coverage-changed-file` should be changed to `min-coverage-changed-files`

**Review by TebaleloS — COMMENTED** [review]
I think this is not lacking anything, just the suggestions that Ladislav made, then it can be approved.

**Review by lsulak — APPROVED** [review]
Looks good now, I quickly scanned through all the CI logs and it seems good, no warnings left

**Pull request merged**